src/main/java/org/jpeek/Report.java

```diff
@@ -228,20 +205,39 @@ public void save(final Path target) throws IOException {
      * Make XML.
      * @return XML
      * @throws IOException If fails
+     * @todo #326:30min Add a test to check whether passing params to
+     *  XSLDocument really works. Currently only C3 metric template
+     *  is known to use parameter named 'ctors'. However C3.xsl is a
+     *  work in progress and has impediments, see #175.
      */
     private XML xml() throws IOException {
         final String name = String.format("metrics/%s.xsl", this.metric);
-        final URL res = this.getClass().getResource(name);
+        final URL res = Report.class.getResource(name);
         if (res == null) {
             throw new IllegalArgumentException(
                 String.format("XSL not found: %s", name)
             );
         }
-        return new XSLDocument(
-            new TextOf(res).asString(),
-            Sources.DUMMY,
-            this.params
-        ).transform(this.skeleton);
+        return new XMLDocument(
+            new Xembler(
+                new Directives()
+                    .xpath("/metric")
+                    .attr(
+                        "xmlns:xsi",
+                        "http://www.w3.org/2001/XMLSchema-instance"
+                    )
+                    .attr(
+                        "xsi:noNamespaceSchemaLocation",
+                        Report.SCHEMA_FILE
+                    )
+            ).applyQuietly(
+                new XSLDocument(
+                    new TextOf(res).asString(),
```

Comment:
Replace that with OO: `                    new TextOf(
                        new ResourceOf(
                            new FormattedText(
                                "org/jpeek/metrics/%s.xsl",
                                this.metric
                            )
                        )
                    ).asString(),`

Reply:
@ilyakharlamov Done.